src/core_modules/capture-core/components/WidgetProfile/DataEntry/FormFoundation/RenderFoundation.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+/* eslint-disable no-underscore-dangle */
+import i18n from '@dhis2/d2-i18n';
+import { capitalizeFirstLetter } from 'capture-core-utils/string/capitalizeFirstLetter';
+import type {
+    CachedProgramTrackedEntityAttribute,
+    CachedTrackedEntityAttribute,
+    CachedTrackedEntityType,
+    CachedOptionSet,
+} from '../../../../storageControllers/cache.types';
+import { RenderFoundation, Section } from '../../../../metaData';
+import { buildDataElement, buildTetFeatureType } from './DataElement';
+import { getProgramTrackedEntityAttributes, getTrackedEntityTypeId } from '../helpers';
+
+const getFeatureType = (featureType: ?string) =>
+    (featureType ? capitalizeFirstLetter(featureType.toLowerCase()) : 'None');
+
+const buildTetFeatureTypeField = (trackedEntityType: CachedTrackedEntityType) => {
+    if (!trackedEntityType) {
+        return null;
+    }
+
+    const featureType = trackedEntityType.featureType;
+    if (!featureType || !['POINT', 'POLYGON'].includes(featureType)) {
+        return null;
+    }
+
+    // $FlowFixMe
+    return buildTetFeatureType(featureType);
+};
+
+const buildTetFeatureTypeSection = async (
+    programTrackedEntityTypeId: string,
+    trackedEntityType: CachedTrackedEntityType,
+) => {
+    const featureTypeField = buildTetFeatureTypeField(trackedEntityType);
+
+    if (!featureTypeField) {
+        return null;
+    }
+
+    const section = new Section((o) => {
+        o.id = programTrackedEntityTypeId;
+        o.name = trackedEntityType?.displayName || '';
+    });
+
+    featureTypeField && section.addElement(featureTypeField);
+    return section;
+};
+
+const buildMainSection = async (
+    trackedEntityType: CachedTrackedEntityType,
+    trackedEntityAttributes: Array<CachedTrackedEntityAttribute>,
+    optionSets: Array<CachedOptionSet>,
+    programTrackedEntityAttributes?: ?Array<CachedProgramTrackedEntityAttribute>,
+) => {
+    const section = new Section((o) => {
+        o.id = Section.MAIN_SECTION_ID;
+        o.name = i18n.t('Profile');
+    });
+
+    if (!programTrackedEntityAttributes?.length) {
+        return null;
+    }
+
+    const featureTypeField = buildTetFeatureTypeField(trackedEntityType);
+    featureTypeField && section.addElement(featureTypeField);
+
+    await buildElementsForSection(programTrackedEntityAttributes, trackedEntityAttributes, optionSets, section);
+    return section;
+};
+
+const buildElementsForSection = async (
+    programTrackedEntityAttributes: ?Array<CachedProgramTrackedEntityAttribute>,
+    trackedEntityAttributes: Array<CachedTrackedEntityAttribute>,
+    optionSets: Array<CachedOptionSet>,
+    section: Section,
+) => {
+    await programTrackedEntityAttributes?.forEach(async (trackedEntityAttribute) => {
```

Comment:
Hey, 
The `asyncForEach` error still persists in Cypress. You can see the error in the below screenshot. 

<img width="2392" alt="Screenshot 2022-01-18 at 16 19 31" src="https://user-images.githubusercontent.com/6817073/150134256-1d37accd-e4e0-4959-858f-b37908a1ec0c.png">

To avoid this strange error I decided to use the `for-of` loop and now all the promises are correctly handled.

Thanks!

Reply:
Investigated a bit and it seems data retrieved from the api is missing the asyncForEach when running the Cypress tests.

Reply:
Makes sense to do `for-of` as you did 👍